qiskit/algorithms/gradients/qfi.py

```diff
@@ -120,13 +120,14 @@ def _run(
             DerivativeType.REAL,
         )
         job = self._qgt.run(circuits, parameter_values, parameters, **options)
-        self._qgt.derivative_type = temp_derivative_type
 
         try:
             result = job.result()
         except AlgorithmError as exc:
             raise AlgorithmError("Estimator job or gradient job failed.") from exc
 
+        self._qgt.derivative_type = temp_derivative_type
```

Comment:
Why is this needed?

Reply:
Thanks. Please check above discussion:
https://github.com/Qiskit/qiskit-terra/pull/9216#issuecomment-1397875203